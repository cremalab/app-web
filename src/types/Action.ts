import { Action } from "redux"

export interface Action<Type extends PropertyKey, Payload>
  extends Action<Type> {
  payload: Payload
}
