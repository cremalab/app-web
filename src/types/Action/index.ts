import { Action as ReduxAction } from "redux"

/**
 * `Action`
 * ---------------------------------------------
 * Extends redux `Action` adding `payload`
 */

export interface Action<Type extends PropertyKey, Payload>
  extends ReduxAction<Type> {
  payload: Payload
}

/**
 * `Action`
 * ------------------
 * Returns a `Tagged<Tag, Data>`
 **/

export const Action = <Type extends PropertyKey, Payload>(
  type: Type,
  payload: Payload,
): Action<Type, Payload> => ({
  type,
  payload,
})
