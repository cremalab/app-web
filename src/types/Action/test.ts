import { Action } from "."

describe("Action", () => {
  it("returns an `Action<Type, Payload>`", () => {
    // Arrange
    const type = "SOME_TYPE"
    const payload = 2

    // Act
    const received = Action(type, payload)

    // Assert
    expect(received).toMatchSnapshot()
  })
})
