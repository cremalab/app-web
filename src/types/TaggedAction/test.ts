import { Action } from "../Action"
import { TaggedAction } from "."

describe("TaggedAction", () => {
  it("returns a `TaggedAction<Action<Type, Payload>>`", () => {
    // Arrange
    const action = Action("TEST_TYPE", 5)

    // Act
    const received = TaggedAction(action)

    // Assert
    expect(received).toMatchSnapshot()
  })
})
