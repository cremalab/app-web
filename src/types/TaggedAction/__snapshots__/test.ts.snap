// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaggedAction returns a \`TaggedAction<Action<Type, Payload>>\` 1`] = `
Object {
  "data": Object {
    "payload": 5,
    "type": "TEST_TYPE",
  },
  "tag": "TEST_TYPE",
}
`;
